text/0000-slice-string-symmetry.md

```diff
@@ -0,0 +1,68 @@
+- Feature Name: `slice_string_symmetry`
+- Start Date: 2015-06-06
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+
+Add some methods that already exist on slices to strings and vice versa.
+Specifically, the following methods should be added:
+
+- `str::chunks`
+- `str::windows`
+- `str::into_string`
+- `String::into_boxed_slice`
+- `<[T]>::subslice_offset`
+
+# Motivation
+
+Conceptually, strings and slices are similar types. Many methods are already
+shared between the two types due to their similarity. However, not all methods
+are shared between the types, even though many could be. This is a little
+unexpected and inconsistent. Because of that, this RFC proposes to remedy this
+by adding a few methods to both strings and slices to even out these two types’
+available methods.
+
+# Detailed design
+
+Add the following methods to `str`, presumably as inherent methods:
+
+- `chunks(&self, n: usize) -> Chunks`: Returns an iterator that yields the
+  *characters* (not bytes) of the string in groups of `n` at a time. Iterator
+  element type: `&str`.
+
+- `windows(&self, n: usize) -> Windows`: Returns an iterator over all contiguous
+  windows of character length `n`. Iterator element type: `&str`.
+
+- `into_string(self: Box<str>) -> String`: Returns `self` as a `String`. This is
+  equivalent to `[T]`’s `into_vec`.
+
+`split_at(&self, mid: usize) -> (&str, &str)` would also be on this list, but
+there is [an existing RFC](https://github.com/rust-lang/rfcs/pull/1123) for it.
+
+Add the following method to `String` as an inherent method:
+
+- `into_boxed_slice(self) -> Box<str>`: Returns `self` as a `Box<str>`,
```

Comment:
That's somewhat related to https://github.com/rust-lang/rfcs/issues/1153, but relying solely on `Into` unfortunately isn't without downsides. For example if I have a `String` and I *know* that I want to go to a `Box<str>` using the `.into()` method would require a type annotation, where a method like this would not. Overall we've been thinking so far that specific conversions (e.g. like this) will have named methods so they can be called ergonomically, but there will also likely be an `Into` implementation to satisfy generic bounds as well. Now that being said, we haven't yet posted the RFC to clarify these conversion conventions.